src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const value = item[key] as string;
+
+    if (result[value]) {
+      result[value] = [...result[value], item];
+    } else {
+      result[value] = [item];
+    }
+  });
```

Comment:
Try to use another array method ( not forEach ) for a solution.
So, you do not need object results and can write the method shortly.